### ISSUE TITLE
archivepath is interpreted differently depending on the value of archiveexploded

### ISSUE THREAD

**Issue by jonsyu1** [cihub/seelog#130]
**Problem:** `archivepath` is interpreted as directory when exploded and file when unexploded.

**Suggestion:** We may not want to break API, but this should be noted in both the [wiki](https://github.com/cihub/seelog/wiki/Receiver-reference).

**Suggestion:** The rolling filewriter should also report the error when the rolling filewriter fails to create the unexploded archive (the user probably specified a directory, since that's the expected value when `archiveexploded` is true.

**Repro steps:**
Configure seelog with `archivepath` pointing to an existing directory and `archiveexploded="false"`
```
        <rollingfile formatid="ourlogStdout" type="date" filename="./var/log/axiom.log" datepattern="2006-01-02_03:04:05" archivetype="gzip" maxrolls="1" archivepath="./var/log" archiveexploded="false"/>
```

**Expected:**
Error creating unexploded archive at "./var/log": is a directory

**Actual:**
Silent failure: no archive created, files continue to be deleted on roll

Conversely, the rolling filewriter should report an error when it points to a file.

**Repro steps:**
Configure seelog with `archivepath` pointing at an existing file and `archiveexploded="true"`
```
        <rollingfile formatid="ourlogStdout" type="date" filename="./var/log/axiom.log" datepattern="2006-01-02_03:04:05" archivetype="gzip" maxrolls="1" archivepath="./var/log/axiom.log.tgz" archiveexploded="true"/>
```

**Expected:**
Error creating exploded archive under `"./var/log/axiom.log.tgz": is a file

**Actual:**
Silent failure: no archives created, files continue to be deleted on roll

**Bottom line:**
This is pretty easily recognizable, but I don't want users to change their `archiveexploded` setting and accidentally lose logs on roll.
